packages/core/core/src/services/document-service/middlewares/defaults/draft-and-publish.ts

```diff
@@ -27,6 +27,17 @@ export const defaultToDraft: Middleware = async (ctx, next) => {
   return next(ctx);
 };
 
+/**
+ * In mutating actions we don't want user to set the publishedAt attribute.
+ */
+export const filterDataPublishedAt: Middleware = async (ctx, next) => {
```

Comment:
Are we totally sure we want to prevent this ? At the controller level, yes, but allowing a programmatic way to set the publishedAt might be useful.

I know we can't throw an error here (or else it would prevent the case of find() -> modify attributes -> update() ), but if we do want to do this, we need to document that it's happening so users aren't confused why it's not working when they set it

(probably should have commented down below on the places this is actually being called, in some of them it makes more sense than others)

Reply:
If we don't filter it, how do we prevent conflicting values between status & publishedAt ? It is true that it could be useful setting this up on an update for example , to update when a document was published, but I am unsure what should be the behaviour here. 

I am opening a ticket to discuss this + other undocumented behaviours of publication state.

Reply:
https://strapi-inc.atlassian.net/browse/CONTENT-2247